src/events/types/inventory/InventoryInteractEvent.ts

```diff
@@ -0,0 +1,46 @@
+import { Player } from '../../../player/Player';
+import { Cancellable } from '../../Cancellable';
+import { EventResult } from '../../EventResult';
+import { InventoryEvent } from './InventoryEvent';
+
+/**
+ * An abstract base class for events that describe an interaction between a HumanEntity and the contents of an Inventory.
+ */
+export class InventoryInteractEvent extends InventoryEvent implements Cancellable {
+
+  /**
+   * Gets the player who performed the click.
+   */
+  public getWhoClicked(): Player {
+    return Player.fromJava(this.toJava().getPlayer());
+  }
+
+  /**
+   * Gets the EventResult of this event.
+   */
+  public getResult(): EventResult {
+    return this.toJava().getResult();
+  }
```

Comment:
I think it should be this, since Java has a `name()` method on enum values that returns a string

```ts
return this.toJava().getResult().name();
```

Reply:
Well, in the way I did EventResult, as per my comment above, this isnt how it works, but I was thinking about changing it to do that